### DISCUSSION TITLE
Retrieve environment variable within building action

### DISCUSSION THREAD

**Discussion by olivierodo** [community/community#26984] [Actions]
<p>Hello, </p>
<p>I’m building a custom action based on Dockerfile and i would like to pass  all the environment variables received to  my container build-in action</p>
<pre><code>name: My new step
      uses: build-action@0.0.1
      env:
        SLACK_WEBHOOK_URL: ${{secrets.SLACK_WEBHOOK_URL}}
        DYNAMIC_VAR: %{{secrets.DYNAMIC_VAR}}
      with:
        path: ./${{ matrix.example_folder }}
</code></pre>
<p>The build-in action looks like :</p>
<pre><code>runs:
  using: 'docker'
  image: 'Dockerfile'
  args:
    - run
    - ${{ inputs.path }}
</code></pre>
<p>How can i get the Env Var SLACK_WEBHOOK_URL, DYNAMIC_VAR and others into my running action container without using defined inputs.</p>
<p>Thank you for your support.</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/olivierodo">@olivierodo</a> , </p>
<p>The step env will be added to the action docker container environment variable , you could just use </p>
<p>$SLACK_WEBHOOK_URL and $DYNAMIC_VAR to get the value in entrypoint.sh</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/0/021f1848ba9b2f119b6ac43cfa9de44fb4b8c037.png" data-download-href="/uploads/short-url/iLzFIWoY3O0wczHBXnOohgeWbR.png?dl=1" title="docker e.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181101063-537b72d9-156d-41e2-bb3b-21c7c2e1e4f6.png" alt="docker e.png" data-base62-sha1="iLzFIWoY3O0wczHBXnOohgeWbR" width="690" height="221" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/0/021f1848ba9b2f119b6ac43cfa9de44fb4b8c037_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">docker e.png</span><span class="informations">1546×497 24.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>My example entrypoint.sh:</p>
<pre><code>#!/bin/sh -l

echo "Hello $1"
time=$(date)
echo ::set-output name=time::$time
echo $SLACK_WEBHOOK_URL
echo $DYNAMIC_VAR</code></pre>
